dist/api_data.py

```diff
@@ -418,8 +414,15 @@ def __ge__(self, other):
         min='50', max='100'),
 ]
 
+readonly_config = [
+    Config('readonly', 'false', r'''
+        the file is read-only. All methods that modify a file are disabled. See @ref
+        readonly for more information''',
+        type='boolean'),
+]
+
 # File metadata, including both configurable and non-configurable (internal)
-file_meta = file_config + [
+file_meta = file_config + readonly_config + [
```

Comment:
This is getting closer. Instead of adding an entire `readonly_config` category, I think you can just add the `readonly` setting into the `file_meta` list. It is the same as `id` and other things that the system manages and the user cannot change. Did you try that?